### DISCUSSION TITLE
Is it a single page webapp (SPA)?

### DISCUSSION THREAD

**Discussion by yipeng-jiang** [zauberzeug/nicegui#566] [Q&A] [answered]
Great framework,  just by glancing it, I like it more than pynecone. One question though, I noticed every time I open a new page using the ui.link. It loads the page completely.  Since it is built on Vue,  why doesn't it behave like a single page web application which routing is handled by the client instead of the server?

**Comment by rodja** [comment, marked as answer]
Thanks @yipeng-jiang. We are happy you like the framework. See https://github.com/zauberzeug/nicegui/discussions/349#discussioncomment-4889399 for a short example on how we do SPAs right now. But I also had the idea that managing the navigation should not be too hard. I just opened the draft pull-request #570 to convey my idea and track remaining issues.

**Reply by rodja** [reply]
The example from #570 is working and merged into main: https://github.com/zauberzeug/nicegui/tree/main/examples/single_page_app
